### PR TITLE
[INFRA-351] enable ML2/OVN and DVR by default

### PR DESCRIPTION
**Pull request by g0rgamesh**

### CODE REVIEW

File: etc/kayobe/kolla.yml [hunk 1]
```diff
@@ -541,7 +541,7 @@ kolla_build_customizations: "{{ kolla_build_customizations_common | combine(koll
 #kolla_enable_neutron:
 #kolla_enable_neutron_agent_ha:
 #kolla_enable_neutron_bgp_dragent:
-#kolla_enable_neutron_dvr:
+kolla_enable_neutron_dvr: yes
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml]: This flag is for neutron OVS DVR.

We need to set `neutron_ovn_distributed_fip: true` in `etc/kayobe/kolla/globals.yml`. See https://docs.openstack.org/kolla-ansible/latest/reference/networking/neutron.html#ovn-ml2-ovn

**Comment by MoteHue** [comment]
I believe that DVR in the ML2/OVS sense is implicitly enabled when using ML2/OVN, so `kolla_enable_neutron_dvr: true` does not need to be set. We will need to set `neutron_ovn_distributed_fip: true` if we want to have the scalable north/south routing. 

Third question here helps to explain a bit: https://docs.openstack.org/neutron/latest/ovn/faq/index.html

File: etc/kayobe/kolla.yml [hunk 2]
```diff
@@ -565,7 +565,7 @@ kolla_build_customizations: "{{ kolla_build_customizations_common | combine(koll
 #kolla_enable_openvswitch:
 #kolla_enable_osprofiler:
 #kolla_enable_outward_rabbitmq:
-#kolla_enable_ovn:
+kolla_enable_ovn: yes
```
Comment (markgoddard) [inline etc/kayobe/kolla.yml:598]: We can now remove this flag from these files:

* `etc/kayobe/environments/aufn-ceph/kolla.yml`
* `etc/kayobe/environments/ci-builder/stackhpc-ci.yml`
* `etc/kayobe/environments/ci-multinode/kolla.yml`

**Review by markgoddard — CHANGES_REQUESTED** [review]
This change needs to go into the stackhpc/zed branch as it's not backward compatible.

Please also add a clear release note with an upgrade section.

**Comment by g0rgamesh** [comment]
> I believe that DVR in the ML2/OVS sense is implicitly enabled when using ML2/OVN, so `kolla_enable_neutron_dvr: true` does not need to be set. We will need to set `neutron_ovn_distributed_fip: true` if we want to have the scalable north/south routing.
> 
> Third question here helps to explain a bit: https://docs.openstack.org/neutron/latest/ovn/faq/index.html

Good catch, thank you.

File: etc/kayobe/kolla.yml (hunk 1 above)
Comment (g0rgamesh) [reply etc/kayobe/kolla.yml]: Thanks, will do.

**Comment by mnasiadka** [comment]
> I believe that DVR in the ML2/OVS sense is implicitly enabled when using ML2/OVN, so `kolla_enable_neutron_dvr: true` does not need to be set. We will need to set `neutron_ovn_distributed_fip: true` if we want to have the scalable north/south routing. 
> 
> 
> 
> Third question here helps to explain a bit: https://docs.openstack.org/neutron/latest/ovn/faq/index.html

To be clear, kolla_enable_neutron_dvr is for ML2/OVS only

**Comment by g0rgamesh** [comment]
> This change needs to go into the stackhpc/zed branch as it's not backward compatible.
> 
> Please also add a clear release note with an upgrade section.

Done.

**Comment by markgoddard** [comment]
> > This change needs to go into the stackhpc/zed branch as it's not backward compatible.
> > Please also add a clear release note with an upgrade section.
> 
> Done.

Still targeting stackhpc/yoga

File: etc/kayobe/kolla.yml (hunk 2 above)
Comment (markgoddard) [reply etc/kayobe/kolla.yml:598]: and the other two?

**Comment by g0rgamesh** [comment]
> > > This change needs to go into the stackhpc/zed branch as it's not backward compatible.
> > > Please also add a clear release note with an upgrade section.
> > 
> > 
> > Done.
> 
> Still targeting stackhpc/yoga

Yep, my bad: https://github.com/stackhpc/stackhpc-kayobe-config/pull/667

File: etc/kayobe/kolla.yml (hunk 2 above)
Comment (g0rgamesh) [reply etc/kayobe/kolla.yml:598]: This branch was created from master, I didn't notice PR was targeting Yoga by default.

**Comment by markgoddard** [comment]
Wrong branch.

**Pull request closed without merging**